moped-editor/src/views/projects/projectView/ProjectComponents/utils/useWatchForPlaceholderLayers.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from "react";
+
+/**
+ * Hook to check map refs to see if placeholder layers that other layers target with beforeId prop
+ * have loaded. This is needed to prevent errors when the placeholder layers are not yet loaded.
+ * @param {Object} mapRef - React ref to mapbox map instance
+ * @returns {Object}
+ * @property {Boolean} hasPlaceHolderLayers - indicating if placeholder layers have loaded
+ * @property {Function} checkIfMapHasPlaceHolderLayers - callback for react-map-gl onLoaded prop
+ * to check if placeholder layers have loaded
+ */
+export const useHavePlaceholderLayersLoaded = (mapRef) => {
+  const [hasPlaceHolderLayers, setHasPlaceHolderLayers] = useState(false);
+
+  const checkIfMapHasPlaceHolderLayers = () => {
+    const hasPlaceholderLayers = Boolean(
+      mapRef?.current?.getLayer("components-placeholder")
```

Comment:
this layer name could be pulled out into a parameter, but this is pretty specific to the components placeholder layer. Maybe a change for down the road - not sure 🤷